sources/web/datalab/static/datalab.js

```diff
@@ -277,6 +277,166 @@ function initializePage(dialog, saveFn) {
   });
 }
 
+// constants for minitoolbar operations
+const CELL_METADATA_COLLAPSED = 'hiddenCell';
+const CELL_METADATA_OUTPUT_COLLAPSED = 'hiddenOutput';
+const COLLAPSE_BUTTON_CLASS = 'fa-eye';
+const UNCOLLAPSE_BUTTON_CLASS = 'fa-eye-slash';
+const COLLAPSE_OUTPUT_BUTTON_CLASS = 'fa-compress';
+const UNCOLLAPSE_OUTPUT_BUTTON_CLASS = 'fa-expand';
+const CLEAR_BUTTON_CLASS = 'fa-minus-square-o';
+const RUN_BUTTON_CLASS = 'fa-play';
+const COLLAPSED_CELL_INNER_HTML = '-- Collapsed cell --';
+
+/**
+ * Collapse entire cell
+ */
+function toggleCollapseCell(cell, hide) {
+  // the cell is currently hidden if its input div is
```

Comment:
It feels odd to me to base this on the view (the hidden state of the input div) instead of the model (the existing cell metadata).

What is the reason for doing it like that?